### PR TITLE
Split cli specs

### PR DESCRIPTION
**Pull request by Morriar**
### Motivation

`cli_spec.rb` is >1200 lines long and will most likely continue to grow as we add CLI commands/options making it hard to scan/edit.

### Implementation

Move each `describe("command")` in its own file with their related code.

Each test now inherits `CliSpec` to get what they need to work.

### CODE REVIEW

File: spec/tapioca/cli_spec.rb [hunk 1]
```diff
@@ -5,1289 +5,52 @@
 require "pathname"
 require "shellwords"
 
-module Contents
```
Comment (Morriar) [inline spec/tapioca/cli_spec.rb:8]: Moved to `generate_spec.rb`

File: spec/tapioca/cli_spec.rb [hunk 2]
```diff
@@ -5,1289 +5,52 @@
...
-    it "prints the correct deprecation message with --cmd" do
-      output = execute("dsl", ["--cmd", "foo"])
-      assert_includes(output, "DEPRECATION: The `-c` and `--cmd` flags will be removed in a future release.")
-    end
-
-    it "doesn't print the correct deprecation message with no flag" do
-      output = execute("dsl")
-      refute_includes(output, "DEPRECATION: The `-c` and `--cmd` flags will be removed in a future release.")
+module Tapioca
+  class CliSpec < Minitest::HooksSpec
```
Comment (Morriar) [inline spec/tapioca/cli_spec.rb:9]: Split Tapioca::CliSpec into a module and a nested class to follow our style guide.

**Comment by Morriar** [comment]
No other changes than the one I commented on Github.

The rest is copy/pasting code into new files.

File: spec/tapioca/cli_spec.rb [hunk 3]
```diff
@@ -5,1289 +5,52 @@
...
-    end
-
-    class Baz::Test
-      def fizz; end
-    end
-  CONTENTS
-end
-
-class Tapioca::CliSpec < Minitest::HooksSpec
-  include TemplateHelper
```
Comment (Morriar) [inline spec/tapioca/cli_spec.rb:62]: Moved to `generate_spec.rb`

File: spec/tapioca/cli/init_spec.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+module Tapioca
+  class InitSpec < CliSpec
+    describe("#init") do
+      it "must create proper files" do
+        FileUtils.rm_f(repo_path / "bin/tapioca")
+        output = execute("init")
+
+        assert_equal(<<-OUTPUT, output)
+      create  sorbet/config
```
[the remark this answers is not in the document]
Comment (Morriar) [reply spec/tapioca/cli/init_spec.rb]: Fixed this in the last commit.

File: spec/tapioca/cli/init_spec.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+          repo_path / "sorbet/config",
+          repo_path / "sorbet/tapioca/require.rb",
+        ])
+
+        output = execute("init")
+
+        assert_equal(<<-OUTPUT, output)
+        skip  sorbet/config
+        skip  sorbet/tapioca/require.rb
+       force  bin/tapioca
```
[the remark this answers is not in the document]
Comment (Morriar) [reply spec/tapioca/cli/init_spec.rb]: Fixed this in the last commit.

**Review by vinistock — APPROVED** [review]

**Pull request merged**